src/deluge/io/midi/midi_device.cpp

```diff
@@ -555,3 +555,17 @@ void MIDIDeviceDINPorts::sendSysex(uint8_t* data, int32_t len) {
 		bufferMIDIUart(data[i]);
 	}
 }
+
+void MIDIDeviceLoopback::writeReferenceAttributesToFile() {}
+
+void MIDIDeviceLoopback::writeToFlash(uint8_t* memory) {}
```

Comment:
Needs to be implemented to learn to commands or midi follow

Reply:
I'm not sure that's actually a useful thing to do, so maybe a popup in midi follow to say that it doesn't work is a better idea

Reply:
Implemented these, I think at least the first one is necessary for getting loopback in MidiDevices.XML for saving MPE settings? Er, still haven't got this stuff mapped in my head enough

Reply:
Yeah MPE settings are saved in a file, global midi learning is saved in flash